### PR TITLE
Added support for Time_Offset parameter in regex parser

### PR DESCRIPTION
**Pull request by JanNemcik**
### What this PR does / why we need it:

Adds support to define Time_Offset parameter on regex parser

### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wenchajun** [comment]
Seems that DCO failed, please commit with a signature.
```
git config --global user.name <name>
git config --global user.email <email>
git commit -s -m "xxx"
```

**Comment by benjaminhuo** [comment]
```shell
git commit -s --amend
git push -f
```

**Comment by JanNemcik** [comment]
@wenchajun @benjaminhuo thanks for help

**Review by wenchajun — APPROVED** [review]

File: apis/fluentbit/v1alpha2/plugins/parser/regex_types.go [hunk 1]
```diff
@@ -39,8 +41,11 @@ func (re *Regex) Params(_ plugins.SecretLoader) (*params.KVs, error) {
 	if re.TimeKeep != nil {
 		kvs.Insert("Time_Keep", fmt.Sprint(*re.TimeKeep))
 	}
+	if re.TimeKeep != nil {
```
Comment (benjaminhuo) [inline apis/fluentbit/v1alpha2/plugins/parser/regex_types.go]: I think you mean `TimeOffset` here, not `TimeKeep`.
`TimeKeep` exists in the lines above

Comment (JanNemcik) [reply apis/fluentbit/v1alpha2/plugins/parser/regex_types.go]: It's fixed in the second commit

File: apis/fluentbit/v1alpha2/plugins/parser/regex_types.go [hunk 2]
```diff
@@ -39,8 +41,11 @@ func (re *Regex) Params(_ plugins.SecretLoader) (*params.KVs, error) {
...
+		kvs.Insert("Time_Keep", fmt.Sprint(*re.TimeKeep))
+	}
 	if re.Types != "" {
-		kvs.Insert("Types", re.Types)
```
Comment (benjaminhuo) [inline apis/fluentbit/v1alpha2/plugins/parser/regex_types.go:43]: We needn't change the `Types` here

File: apis/fluentbit/v1alpha2/plugins/parser/regex_types.go (hunk 1 above)
Comment (benjaminhuo) [reply apis/fluentbit/v1alpha2/plugins/parser/regex_types.go]: Got it

**Comment by benjaminhuo** [comment]
@JanNemcik Thank you for the contribution!

**Pull request merged**